core-services/documentTemplate/src/main/java/org/silverpeas/core/documenttemplate/DocumentTemplate.java

```diff
@@ -0,0 +1,220 @@
+   * @param language the locale into which the label should be provided.
+   * @return true of a translation exists, false otherwise.
+   */
+  public boolean existNameFor(final String language) {
+    return json.getNameTranslations().containsKey(language);
+  }
+
+  /**
+   * Gets the name of the document template.
+   * @param language the locale into which the label should be provided. A default one is
+   * provided if none retrieved with given language.
+   * @return a string representing a name.
+   */
+  public String getName(final String language) {
+    return json.getNameTranslations().get(language);
+  }
+
+  /**
+   * Sets the name of the document template.
+   * @param name a name.
+   * @param language the locale of the given name.
+   */
+  public void setName(final String name, final String language) {
+    json.getNameTranslations().put(language, name);
+  }
+
+  /**
+   * Indicates if a name translation exists for the given language.
+   * @param language the locale into which the label should be provided.
+   * @return true of a translation exists, false otherwise.
+   */
+  public boolean existDescriptionFor(final String language) {
+    return json.getDescriptionTranslations().containsKey(language);
+  }
+
+  /**
+   * Gets the description of the document template.
+   * @param language the locale into which the description should be provided. A default one is
+   * provided if none retrieved with given language.
+   * @return a string representing a description.
+   */
+  public String getDescription(final String language) {
+    return json.getDescriptionTranslations().get(language);
+  }
+
+  /**
+   * Sets the description of the document template.
+   * @param description a description.
+   * @param language the locale of the given description.
+   */
+  public void setDescription(final String description, final String language) {
+    json.getDescriptionTranslations().put(language, description);
+  }
+
+  /**
+   * Gets the file extension of the document template.
+   * @return a string (without the file extension separator).
+   */
+  public String getExtension() {
+    return extension;
+  }
+
+  /**
+   * Sets the file extension of the document template.
+   * @param extension a string (without the file extension separator).
+   */
+  public void setExtension(final String extension) {
+    this.extension = extension;
+  }
+
+  /**
+   * Gets the viewer context of the documen template.
+   * <p>
+   *   Useful for {@link DocumentView} and {@link Preview} services.
+   * </p>
+   * @param language the language into which the {@link ViewerContext} is expected.
+   * @return a {@link ViewerContext} instance.
+   */
+  public ViewerContext getViewerContext(final String language) {
+    return new ViewerContext(getId(), DOCUMENT_TEMPLATE, getName(language),
+        getContentFilePath().toFile(), I18n.get().getDefaultLanguage())
+        .withUniqueDocumentId(language + "-" + getId());
```

Comment:
Because the specified language is one of a user, whereas the default language used is about a content language.
In case of document templates, content language is not handled, so the default one is here given.

Reply:
ok

Reply:
To avoid misreading and misunderstanding, I think one day I18n and `I18nHelper` will have to be renamed respectively to `ContentI18n` and `ContentI18nHelper`.